.github/workflows/publish.yml

```diff
@@ -0,0 +1,73 @@
+name: Deploy Next.js site to Pages
```

Comment:
Name is more or less fine, it was the default for this workflow and is sufficiently descriptive, but it is not matching the file name: it was sufficiently descriptive on the other repo, because the repo already had a definite scope, but not in this one (that is generic for workflows).

Change file name to something closer to the name, like `publish-next-on-ghpages.yaml`.